Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp

```diff
@@ -0,0 +1,182 @@
+///
+/// @tparam detector_element_t is the actual type of the detector
+/// element, each derivative of a TelescopeDetectorElement can be used
+///
+/// @param gctx is the detector element dependent geometry context
+/// @param detectorStore is the store for the detector element
+/// @param matDecorator is an optional decorator for the material
+template <typename detector_element_t>
+std::unique_ptr<const Acts::TrackingGeometry> buildDetector(
+    const typename detector_element_t::ContextType& gctx,
+    std::vector<std::shared_ptr<detector_element_t>>& detectorStore,
+    const std::vector<double>& longShifts,
+    const std::vector<double>& tranShifts, const std::vector<double>& boundary,
+    double thickness, Acts::BinningValue binValue = Acts::BinningValue::binZ) {
+  using namespace Acts::UnitLiterals;
+
+  // The rectangle bounds
+  // @todo make the bound size configurable
+  const auto rBounds = std::make_shared<const Acts::RectangleBounds>(
+      Acts::RectangleBounds(boundary[0], boundary[1]));
+
+  // Material of the surfaces
+  Acts::Material silicon = Acts::Material::fromMolarDensity(
+      9.370_cm, 46.52_cm, 28.0855, 14, 2.329_g / 1_cm3);
+  Acts::MaterialSlab matProp(silicon, thickness);
+  const auto surfaceMaterial =
+      std::make_shared<Acts::HomogeneousSurfaceMaterial>(matProp);
+
+  // Construct the rotation
+  // This assumes the binValue is binX, binY or binZ
+  Acts::RotationMatrix3 rotation = Acts::RotationMatrix3::Identity();
+  if (binValue == Acts::BinningValue::binX) {
+    rotation.col(0) = Acts::Vector3(0, 0, -1);
+    rotation.col(1) = Acts::Vector3(0, 1, 0);
+    rotation.col(2) = Acts::Vector3(1, 0, 0);
+  } else if (binValue == Acts::BinningValue::binY) {
+    rotation.col(0) = Acts::Vector3(1, 0, 0);
+    rotation.col(1) = Acts::Vector3(0, 0, -1);
+    rotation.col(2) = Acts::Vector3(0, 1, 0);
+  }
+
+  // Set translation vectors
+  size_t nLayers = longShifts.size();
+  std::vector<Acts::Vector3> translations;
+  translations.reserve(nLayers);
+  // The transverse shift is the same for all planes
+  Acts::Vector3 tshift(0, 0, 0);
+  if (binValue == Acts::BinningValue::binX) {
+    tshift.y() = tranShifts[0];
+    tshift.z() = tranShifts[1];
+  } else if (binValue == Acts::BinningValue::binY) {
+    tshift.x() = tranShifts[0];
+    tshift.z() = tranShifts[1];
+  } else {
+    tshift.x() = tranShifts[0];
+    tshift.y() = tranShifts[1];
+  }
+  for (const auto& pos : longShifts) {
+    // The longidutinal shift
+    Acts::Vector3 lshift(0, 0, 0);
+    // This assumes the binValue is binX, binY or binZ
+    lshift[binValue] = pos;
+    translations.push_back(tshift + lshift);
+  }
+
+  // Construct surfaces
+  std::vector<std::shared_ptr<const Acts::Surface>> surfaces;
+  surfaces.reserve(nLayers);
+  unsigned int i;
+  for (i = 0; i < nLayers; i++) {
+    Acts::Transform3 trafo(Acts::Transform3::Identity() * rotation);
+    trafo.translation() = translations[i];
```

Comment:
I always find it easier to understand the transformation if build directly as
```cpp
Acts::Transform3 trafo = Acts::Translation3(translations[i]) * rotation;
```
That way it is clear that it is rotation first, then translation. You might also be able to do
```cpp
Acts::Transform3 trafo = rotation * Acts::Translation3(u, v, w);
```
in which case you do not need to rotate the translations in the local frame first.

Reply:
Exactly. Done.